src/main/resources/assets/js/models/Router.js

```diff
@@ -0,0 +1,33 @@
+define([
+  "Backbone"
+], function(Backbone) {
+  "use strict";
+
+  var noop = function () {};
+
+  return Backbone.Router.extend({
+
+    routes: {
+      "": "apps",
+      "apps": "apps",
+      "deployments": "deployments",
+      "app/:appid(/:view)": "app",
```

Comment:
Did it to apps(/:appid)(/:view) and routeApps(params).
To handle all changes in one function is difficult, because the view state is changing in the AppModalComponent and not in Marathon.